### PR TITLE
[sfp-refactoring] Initial support for CMIS application initialization

### PR DESCRIPTION
**Pull request by ds952811**
[sfp-refactoring] Quanta IX9: migrate to the new sfp-refactoring framework

Migrate the Quanta IX9 to the new sfp-refactoring framework

Signed-off-by: Dante Su <dante.su@broadcom.com>

#### Why I did it
Migrate the Quanta IX9 to the new sfp-refactoring framework

#### How I did it
Update the sfp.py of Quanta IX9 to use the new sfp-refactoring framework

#### How to verify it
1. Bring up Quanta IX9 with the default configuration and CMIS 4.0/5.0 QSFPDD modules
2. Issue 'sudo sfputil show eeprom --dom' to verify both the static EEPROM and DOM information
3. Issue 'sudo sfputil show error-status -hw' to verify error states

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

#### Description for the changelog

<table>
<tr>
<th>Repo</th>
<th>PR title</th>
<th>State</th>
</tr>
<tr>
<td>SONiC</td>
<td><a href='https://github.com/Azure/SONiC/pull/876'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td></td>
</tr>
<tr>
<td>sonic-buildimage</td>
<td><a href='https://github.com/Azure/sonic-buildimage/pull/9157'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td></td>
</tr>
<tr>
<td>sonic-platform-common</td>
<td><a href='https://github.com/Azure/sonic-platform-common/pull/219'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td></td>
</tr>
<tr>
<td>sonic-platform-daemon</td>
<td><a href='https://github.com/Azure/sonic-platform-daemons/pull/217'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td></td>
</tr>
<td>sonic-utilities</td>
<td><a href='https://github.com/Azure/sonic-utilities/pull/1855'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td></td>
</table>

### CODE REVIEW

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
which sfp refactor framework are you referring to? is it arista one or broadcom one?

**Comment by ds952811** [comment]
> which sfp refactor framework are you referring to? is it arista one or broadcom one?

It's Broadcom, but the new sfp-refactoring is still in-progress, and should be ready by this Friday

File: platform/broadcom/sonic-platform-modules-quanta/ix9-32x/sonic_platform/sfp.py [hunk 1]
```diff
@@ -321,56 +69,22 @@ def __init__(self, sfp_index, sfp_type):
...
+    def get_port_type(self):
+        """
+        Retrieves the port/cage type of this SFP
 
-    def __is_host(self):
-        return os.system(self.HOST_CHK_CMD) == 0
+        Returns:
+            A string, the port/cage type of this SFP
+        """
+        return self.SFP_PORT_TYPE_QSFPDD
```
Comment (mikeberesford) [inline platform/broadcom/sonic-platform-modules-quanta/ix9-32x/sonic_platform/sfp.py:84]: Shouldn't this (at least ideally) be based on byte 00 from the plugged in module?

Comment (ds952811) [reply platform/broadcom/sonic-platform-modules-quanta/ix9-32x/sonic_platform/sfp.py:79]: This is the cage type of the port, not the media type fetched from the EEPROM of the attached transceiver.

Comment (mikeberesford) [reply platform/broadcom/sonic-platform-modules-quanta/ix9-32x/sonic_platform/sfp.py:79]: ack, thanks.

**Pull request closed without merging**